### ISSUE TITLE
[aws-codebuild] - Typescript, missing BuildSpec type definition

### ISSUE THREAD

**Issue by alanbo** [aws/aws-cdk#8872]
There is no code completions/validation for `codebuild.BuildSpec.fromObject`. Typescript definition of it has an Indexable Type instead of proper definition:

```
export declare abstract class BuildSpec {
    static fromObject(value: {
        [key: string]: any;
    }): BuildSpec;
    /**
     * Use a file from the source as buildspec
     *
     * Use this if you want to use a file different from 'buildspec.yml'`
     */
    static fromSourceFilename(filename: string): BuildSpec;
    /**
     * Whether the buildspec is directly available or deferred until build-time
     */
    abstract readonly isImmediate: boolean;
    protected constructor();
    /**
     * Render the represented BuildSpec
     */
    abstract toBuildSpec(): string;
}
```

The main reason to use CDK is having code completions/validation to avoid error - in this case, it is missing. 

### Reproduction Steps
use `BuildSpec.fromObject` of `@aws-cdk/aws-codebuild`. 

--- 

This is :bug: Bug Report

**Comment by skinny85** [comment]
Hey @alanbo ,

this is not a bug, it's a feature :). The `BuildSpec.fromObject()` is not typed, so that you're not blocked by us from using all of the new Buildspec features that CodeBuild releases (which is actually quite frequent!).

We have an issue about providing a better, type-safe alternative: https://github.com/aws/aws-cdk/issues/2039 , probably something like `BuildSpe.version02(...)`. If you'd like to take a stab at implementing it, I would love to help you out 🙂.

Thanks,
Adam
